### PR TITLE
Pull request templates

### PR DESCRIPTION
**Pull request by alanmcruickshank**
Following on from issue template, this is my stab at PR templates. In particular I'd like to prompt people to update the `CHANGELOG.md` more diligently to make the release cycle easier.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/bug_fix.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+name: Bug Fix/Demonstration
+about: This fixes or demonstrates a bug in SQLFluff
+labels: bug
+---
+
+-- Firstly, thanks for fixing a bug! Secondly, please make sure your PR includes a test case demonstrating that the bug is fixed. Test cases should be as minimal as possible to show it's fixed but also be representative of the case the original reporter of the bug posted. If this PR is just a failing test case, please file it as a `draft` PR or use pytest `xfail` to mark the case as expected to fail. --
```
Comment (barrywhart) [inline .github/PULL_REQUEST_TEMPLATE/bug_fix.md:7]: Could it be helpful to document the more common types of test cases and how to create them? I'm mainly thinking of the standalone `.sql`/`.yaml` parser test cases and the `.yml` lint rule test cases, but there may be others.

File: .github/PULL_REQUEST_TEMPLATE/bug_fix.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+- [ ] This PR includes test cases to demonstrate the fix.
+- [ ] This PR updates the [CHANGELOG.md](https://github.com/sqlfluff/sqlfluff/blob/master/CHANGELOG.md).
+
+### What was the root cause of this bug?
+...
+
+### Are there any other side effects of this fix?
+...
+
+### How comprehesive is the test coverage of this fix?
```
Comment (barrywhart) [inline .github/PULL_REQUEST_TEMPLATE/bug_fix.md:24]: Typo: comprehesive -> comprehensive

**Review by barrywhart — CHANGES_REQUESTED** [review]

**Review by pwildenhain — APPROVED** [review]
Great work alan 💰🎖️😎

**Comment by dmateusp** [comment]
excited about those templates too!

**Pull request merged**